collector/collector/report_handler.py

```diff
@@ -191,15 +237,18 @@ def collector_post_handler():
     else:
         external = False
 
-    try:
-        # prefer UTF-8, if possible
-        payload = request.data.decode('utf-8')
-    except UnicodeError:
-        # fallback to Latin-1, since it accepts all byte values
-        payload = request.data.decode('latin-1')
+    # Handles payload depending on payload version and mime type
+    payload, mime_type = process_payload()
+    attachment = None
 
-    if classification == "org.clearlinux/mce/corrected" and "THERMAL" in payload:
-        classification = "org.clearlinux/mce/thermal"
+    # Interpret payload_format_version
+    if BINARY_ATTACHMENT[0] <= int(payload_format_version) <= BINARY_ATTACHMENT[1]:
```

Comment:
Can you elaborate on the significance of `payload_format_version` having a value in the range of 200 to 300? What happens if the value is outside that range?

Reply:
A version between 200 and 300 means that the payload is an file uploaded as a multipart/form-data element in the body with key "payload" and such element should be saved as a file in the quarantine directory and a metadata record for it should be created in the DB.  Anything outside the range will be treated as it is treated now (text in body).

The reason is not a single value and is a range is because I would like to have "room" to specify more characteristics of the attachment in the version i.e.  200 is an attachment, 201 is a zipped attachment, etc. This is not the same as "how this record should be parsed", which is a different question and personally I think should be decided based on category (which probe send the info).

Reply:
Ah, okay. The original idea was for each probe to define its own payload_format_version, due to incompatible changes it the payload format that it sends. Very similar to the intended use of record_format_version, but except for the probes control the value instead.

Maybe adding a new field would be better? Say, "record_type" to indicate whether it is plain text only, has an attachment, etc? That said, telemetrics-backend has never used payload_format_version before now, so we can redefine its semantics.